### PR TITLE
Tests for DataCollection

### LINKED ISSUE #285: Overwrite protection for DataCollection does not work
DataCollections should prevent items to be overwritten by items with the same key. However, it seems that this does not work in practise. There are no tests covering the code to check for existing keys. See #286

### PR DESCRIPTION
**Pull request by psomhorst**
Tests for `DataCollection`s were missing. Three issues with `DataCollection` or data loading were found and fixed in this PR. 

The tests cover 95% of the code in datacollection.py. The `select_by_time()` method is not tested for `DataCollection[IntervalData]` because `select_by_time()` is not finalised for `IntervalData`. Created #306 to add this test when #198 is fixed.

Closes #285 #286.

### CODE REVIEW

File: tests/test_datacollection.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+    dc.add(data_object_2, data_object_3)
+    assert "label 2" in dc
+    assert "label 3" in dc
+    assert len(dc) == 3
+
+    with pytest.raises(KeyError):
+        dc.add(data_object_1_b)
+
+    dc.add(data_object_1_b, overwrite=True)
```
Comment (JulietteFrancovich) [inline tests/test_datacollection.py:100]: Wouldn't it make sense to assert whether this overwrite acts as expected (check length of dc, check whether dc['label 1'] is now data_object_1_b)? To me it seems weird to do this overwrite here but not add any assertions but maybe I am missing something.

File: tests/test_datacollection.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+    dc_1_concat = dc_1_a.concatenate(dc_1_b)
+    assert np.array_equal(dc_1_concat["label 1"].time, time)
+    assert np.array_equal(dc_1_concat["label 1"].values, values)
+
+    with pytest.raises(ValueError, match="Keys don't match"):
+        dc_1_a.concatenate(dc_2)
+
+    with pytest.raises(ValueError, match=r"(.*?) \(b\) starts before (.*?) \(a\) end"):
+        dc_1_b.concatenate(dc_1_a)
```
Comment (JulietteFrancovich) [inline tests/test_datacollection.py:177]: Perhaps also test behavior of concatenate with overlapping time ranges

File: tests/test_datacollection.py (hunk 1 above)
Comment (JulietteFrancovich) [reply tests/test_datacollection.py:100]: The same for the last line of test_check_item

File: tests/test_datacollection.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    values1 = rng.random(time1.shape)
+    values2 = rng.random(time2.shape)
+
+    data_object_1 = create_data_object("label 1", time=time1, values=values1)
+    data_object_2 = create_data_object("label 2", time=time2, values=values2)
+
+    dc = DataCollection(ContinuousData)
+    dc.add(data_object_1, data_object_2)
+
+    s1 = dc.select_by_time(20, 80)
```
Comment (JulietteFrancovich) [inline tests/test_datacollection.py:193]: Perhaps also test what happens if start and end of select_by_time are the same?

**Review by JulietteFrancovich — COMMENTED** [review]
@psomhorst I have reviewed the tests for data collection. I think in general it is very clear and comprehensive. I have a few small questions/ suggestions.

File: tests/test_datacollection.py (hunk 3 above)
Comment (psomhorst) [reply tests/test_datacollection.py:193]: This is a good suggestion. However, `select_by_time` currently is not in a great state and will be rewritten partly.
Therefore, I'll add is as TODO.

File: tests/test_datacollection.py (hunk 1 above)
Comment (psomhorst) [reply tests/test_datacollection.py:100]: `test_check_item` does not change `dc`, so there is nothing to assert there.

**Review by JulietteFrancovich — APPROVED** [review]

**Pull request closed without merging**